barman/cli.py

```diff
@@ -52,6 +56,156 @@
+                dest = name
+                if aliases:
+                    dest += " (%s)" % ",".join(aliases)
+                sup = super(AliasedSubParsersAction._AliasedPseudoAction, self)
+                sup.__init__(option_strings=[], dest=dest, help=help)
+
+        def add_parser(self, name, **kwargs):
+            aliases = kwargs.pop("aliases", [])
+            parser = super(AliasedSubParsersAction, self).add_parser(name, **kwargs)
+
+            # Make the aliases work.
+            for alias in aliases:
+                self._name_parser_map[alias] = parser
+            # Make the help text reflect them, first removing old help entry.
+            if "help" in kwargs:
+                help = kwargs.pop("help")
+                self._choices_actions.pop()
+                pseudo_action = self._AliasedPseudoAction(name, aliases, help)
+                self._choices_actions.append(pseudo_action)
+
+            return parser
+
+    # override argparse to register new subparser action by default
+    _ActionsContainer.__init__ = AliasedSubParsersAction._containerInit
+
+
+class OrderedHelpFormatter(HelpFormatter):
+    def _format_usage(self, usage, actions, groups, prefix):
+        for action in actions:
+            if not action.option_strings:
+                action.choices = OrderedDict(sorted(action.choices.items()))
+        return super(OrderedHelpFormatter, self)._format_usage(
+            usage, actions, groups, prefix
+        )
+
+
+p = ArgumentParser(
+    epilog="Barman by EnterpriseDB (www.enterprisedb.com)",
+    formatter_class=OrderedHelpFormatter,
+)
+p.add_argument(
+    "-v",
+    "--version",
+    action="version",
+    version="%s\n\nBarman by EnterpriseDB (www.enterprisedb.com)" % barman.__version__,
+)
+p.add_argument(
+    "-c",
+    "--config",
+    help="uses a configuration file "
+    "(defaults: %s)" % ", ".join(barman.config.Config.CONFIG_FILES),
+    default=SUPPRESS,
+)
+p.add_argument(
+    "--color",
+    "--colour",
+    help="Whether to use colors in the output",
+    choices=["never", "always", "auto"],
+    default="auto",
+)
+p.add_argument(
+    "--log-level",
+    help="Override the default log level",
+    choices=list(get_log_levels()),
+    default=SUPPRESS,
+)
+p.add_argument("-q", "--quiet", help="be quiet", action="store_true")
+p.add_argument("-d", "--debug", help="debug output", action="store_true")
+p.add_argument(
+    "-f",
+    "--format",
+    help="output format",
+    choices=output.AVAILABLE_WRITERS.keys(),
+    default=output.DEFAULT_WRITER,
+)
+
+subparsers = p.add_subparsers(dest="command")
+
+
+def argument(*name_or_flags, **kwargs):
+    """Convenience function to properly format arguments to pass to the
+    command decorator.
+    """
+
+    # Remove the completer keyword argument from the dictionary
+    completer = kwargs.pop("completer", None)
+    return (list(name_or_flags), completer, kwargs)
+
+
+def command(args=[], parent=subparsers, cmd_aliases=[]):
```

Comment:
We should avoid using mutables as defaults for `args` and `cmd_aliases` here, as we did with `args=None` earlier in the PR.

Reply:
Yes - we'd also need to test for `None` and assign an empty list for both parameters at the top of the function.